src/DI/MultiplierExtension.php

```diff
@@ -6,31 +6,47 @@
 use Contributte\FormMultiplier\Multiplier;
 use Nette;
 use Nette\DI\CompilerExtension;
+use Nette\Schema\Expect;
+use Nette\Schema\Processor;
+use Nette\Schema\Schema;
 
 class MultiplierExtension extends CompilerExtension
 {
 
-	/** @var string[] */
-	public $defaults = [
-		'name' => 'addMultiplier',
-	];
+	public function getConfigSchema(): Schema
+	{
+		return Expect::structure([
+			'name' => Expect::string()->default('addMultiplier'),
+		]);
+	}
 
 	public function beforeCompile(): void
 	{
 		$builder = $this->getContainerBuilder();
 
-		$resultDefinition = $builder->getDefinition('latte.latteFactory')
-			->getResultDefinition();
+		$latteFactoryDefinition = $builder->getDefinition('latte.latteFactory');
+		if ($latteFactoryDefinition instanceof Nette\DI\Definitions\FactoryDefinition === false) {
+			throw new Nette\DI\InvalidConfigurationException(
+				sprintf(
+					'latte.latteFactory service definition must be of type %s, not %s',
+					Nette\DI\Definitions\FactoryDefinition::class,
+					get_class($latteFactoryDefinition)
+				)
+			);
+		}
+
+		$resultDefinition = $latteFactoryDefinition->getResultDefinition();
 
 		$resultDefinition->addSetup(MultiplierMacros::class . '::install(?->getCompiler())', ['@self']);
 	}
 
 	public function afterCompile(Nette\PhpGenerator\ClassType $class): void
 	{
-		$init = $class->getMethods()['initialize'];
-		$config = $this->validateConfig($this->defaults);
+		$processor = new Processor();
+		$config = $processor->process($this->getConfigSchema(), $this->getConfig());
```

Comment:
Did you try `$this->config`?

Reply:
Schema is there because of the default value and possible addition of more configuration options. If you think more direct approach would be better, I can change that to something like

```php
$methodName = $this->config ?? 'addMultiplier';
// ...
$init->addBody(Multiplier::class . '::register(?);', [$methodName]);
```

but as fair as I understand it correctly, that would not validate the config at all.

Reply:
@jankonas f3l1x thought something like

```php
$config = $this->getConfig();

$init->addBody(Multiplier::class . '::register(?);', [$config->name]);
```

Reply:
You were right, validation is called automatically by Nette.